corehq/apps/locations/templates/locations/manage/locations.html

```diff
@@ -54,107 +56,56 @@
-        var modal_context;
+    $(function() {
+        var model = new LocationTreeViewModel(hierarchy);
+        $('#location_tree').koApplyBindings(model);
+        model.load(locs);
+    });
+
+
+    {% if request|toggle_enabled:'LOCATION_SEARCH' %}
+
+    var enableSelect = function() {
 
-        function delete_fn() {
-            if (modal_context.count == modal_context.signOff()) {
-                $(button).disableButton();
-                $.ajax({
-                    type: 'DELETE',
-                    url: loc_delete_url(loc_id),
+        $('#location_search_select').select2({
+            ajax: {
+                url: '{% url 'location_search' domain %}',
                     dataType: 'json',
-                    error: function (response, status, error) {
-                        alert_user(delete_error_message, "warning");
-                        $(button).enableButton();
+                    data: function (params) {
+                        console.log(params);
+                        return {
+                            q: params.term,
+                            page_limit: 10,
+                            page: params.page,
+                        };
                     },
-                    success: function (response) {
-                        if (response.success){
-                            alert_user(delete_success_message({"name": name}), "success");
-                            remove_elements_after_action(button);
-                        }
-                        else {
-                            alert_user(response.message, "warning");
-                            $(button).enableButton();
-                        }
+                    processResults: function (data, params) {
+                        var more = data.more || (params.page * 10) < data.total;
+                        return {results: data.results, more: more};
                     }
-                });
-                $(delete_location_modal).modal('hide');
-                hqImport('analytix/js/google').track.event('Organization Structure', 'Delete')
-            }
-        }
-
-        $.ajax({
-            type: 'GET',
-            url: loc_descendant_url(loc_id),
-            dataType: 'json',
-            success: function (response) {
-                modal_context = {
-                    "name": name,
-                    "loc_id": loc_id,
-                    "delete_fn": delete_fn,
-                    "count": response.count,
-                    "signOff": ko.observable('')
-                };
-                ko.cleanNode(delete_location_modal);
-                ko.applyBindings(modal_context, delete_location_modal);
-                $(delete_location_modal).modal('show');
-            }
-        });
-    };
+                },
+            });
+        };
 
-    remove_elements_after_action = function(button) {
-        $(button).closest('.loc_section').remove();
-    };
+        var reloadSelect = function() {
```

Comment:
This is a pretty generic name for a function in the global namespace! Maybe rename this to something like `reloadLocationSearchSelect`. If you end up being able to get this out of the global namespace with a refactor, then you can change it back to being called something shorter.

Reply:
(Same with `enableSelect`.)